uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java

```diff
@@ -298,7 +298,7 @@ public int getHeaderRowCount() {
 
     @Override
     public void setHeaderRowCount(final int headerRowCount) {
-        if (!(headerRowCount > 0)) {
+        if (!(headerRowCount >= 0)) {
```

Comment:
I slightly think we could remove this `setHeaderRowCount` method completely as the `getHeaderRowCount` is kind of independent and uses `headerRowCount` private field just as some default return value for case whn no column are specified or they have no header metadata , is this correct?

Probably we won't risk regressions so keep this method, however please update javadoc in `GridData` interface.